src/components/calcite-panel/calcite-panel.scss

```diff
@@ -42,32 +43,52 @@
 }
 
 .header {
-  align-items: center;
+  align-items: stretch;
   display: flex;
   justify-content: space-between;
+  flex: 0 1 var(--calcite-app-panel-min-header-height);
   position: relative;
   z-index: 2;
   border-bottom: 1px solid var(--calcite-app-border);
   width: 100%;
 }
 
-.header-content {
+.header-middle-content {
+  display: flex;
+  flex-flow: column;
   overflow: hidden;
-  padding: var(--calcite-app-cap-spacing-quarter) var(--calcite-app-side-spacing);
+  padding: var(--calcite-app-cap-spacing) var(--calcite-app-side-spacing);
 }
 
-.header-leading-content + .header-content {
+.header-leading-content,
+.header-trailing-content {
+  display: flex;
+}
+
+.header-leading-content + .header-middle-content {
   padding-left: var(--calcite-app-side-spacing-half);
 }
 
-slot[name="header-content"]::slotted(.heading) {
+slot[name="header-content"]::slotted(.heading),
+slot[name="header-summary"]::slotted(.summary) {
+  display: block;
   overflow: hidden;
   text-overflow: ellipsis;
   white-space: nowrap;
   width: 100%;
 }
 
+slot[name="header-content"]::slotted(.heading) {
```

Comment:
these seem pretty brittle. "A slot that has a class of heading"?

Reply:
So if a user doesn't supply the heading class this won't be styled correctly. Or if the class is renamed in the flow-item, this would break.